### PR TITLE
ETCD as a configuration provider #43

### PR DESCRIPTION
**Pull request by GGXXLL**

### CODE REVIEW

**Comment by Reasno** [comment]
pending v0.5

File: c.go [hunk 1]
```diff
@@ -89,6 +91,19 @@ func WithYamlFile(path string) (CoreOption, CoreOption) {
 		WithConfigWatcher(watcher.File{Path: path})
 }
 
+// WithRemoteYamlFile is a two-in-one coreOption. It uses the remote file on etcd as the
```
Comment (Reasno) [inline c.go:94]: file -> key

File: c.go [hunk 2]
```diff
@@ -89,6 +91,19 @@ func WithYamlFile(path string) (CoreOption, CoreOption) {
...
+// source of configuration, and watches the change of that file for hot reloading.
+func WithRemoteYamlFile(path string, cfg clientv3.Config) (CoreOption, CoreOption) {
+	r := remote.Provider(path, &cfg)
+	return WithRemote(r)
+}
+
+// WithRemote is a two-in-one coreOption. It uses the etcd's file as the
```
Comment (Reasno) [inline c.go:101]: file->key

File: config/remote/remote.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+				return err
+			}
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+	}
+}
+
+// Put add values to remote config file on etcd.
+func (r *Remote) Put(val string) error {
```
Comment (Reasno) [inline config/remote/remote.go:79]: It is unnecessary to provide this API. I intend to keep API surface as small as possible.

Comment (Reasno) [reply config/remote/remote.go:79]: I am ok with moving this to remote_test.go package or making it private.

**Review by Reasno — CHANGES_REQUESTED** [review]

File: config/remote/remote.go (hunk 3 above)
Comment (GGXXLL) [reply config/remote/remote.go:79]: LGTM, i will do it.

**Review by Reasno — APPROVED** [review]

**Pull request merged**